### DISCUSSION TITLE
difference with kamalshkeir/kago

### DISCUSSION THREAD

**Discussion by manson** [pocketbase/pocketbase#219] [Q&A] [answered]
Could you please tell the differences with package https://github.com/kamalshkeir/kago? Looking here and there I see almost the same functionality, but I may be wrong. You as author could make a quick look and see the diffs and similarities or maybe see something useful for yourself (or not) and tell us... Thanks

**Comment by ganigeorgiev** [comment, marked as answer]
Hi, I'm not familiar with this project.

I may have a more detailed look at it some weekend, but at first glance, based on the readme examples, it looks like a general purpose framework so I guess it covers some of the functionalities that PocketBase provide.

In terms of differences - PocketBase approaches the development process UI first and allows to extend with custom go code if the existing functionalities are not enough, while kago seems to be code first - requiring the developers to write every endpoint handler on their own (or generate it via console command).

It seems like an interesting project, so if you prefer more Django/Rails/Laravel/Symfony style of development then I guess you could give it a try.

**Reply by manson** [reply]
There is some admin autogenerated UI as they say, I didn't test it though as I tested your project. I like both styles - UI first and code-first, it depends. But I would like you to pay attention to their migration engine. It looks very easy and comfortable for developer (just migrate model command and that's it). I don't like migrations parts, so because of this i use almost all the time the documents based databases, but this way as in Kago implemented, eliminates the most part of difficulties... This was almost the single stop-point that didn't let me consider the pocketbase for  usage for some future project... :-)

**Reply by ganigeorgiev** [reply]
Auto generated migration files for collections (see https://github.com/pocketbase/pocketbase/discussions/175) and import/export (for both data and schema) is in the roadmap and will be handled as part of issue https://github.com/pocketbase/pocketbase/issues/48 after releasing the Dart SDK.
